pkg/reconciler/ingress/controller.go

```diff
@@ -111,7 +112,16 @@ func NewController(
 		DeleteFunc: func(obj interface{}) {
 			vs, ok := obj.(*v1alpha3.VirtualService)
 			if ok {
-				statusProber.CancelVirtualServiceProbing(vs)
+				// The VirtualService controlling the Gateways always has the same name with the parent Ingress.
+				// We use that knowledge here to cancel the probing when VirtualService is deleted.
```

Comment:
So I noticed the same name bit when I was working on the Contour controller (and the original separation), and so I made the Contour controller's version of this just wire it up directly to Ingress-deleted events instead of VirtualService.  What I like about this is that it lets us reorient the `CancelIngressProbing` in a way that it could avoid all of this thunk logic.

e.g. the goal should be:
```go
ingressInformer.Informer().AddEventHandler(cache.ResourceEventHandlerFuncs{
     DeleteFunc: statusProber.CancelIngressProbing,
})
```

Reply:
Also, I feel like the DeleteFunc should be using: https://github.com/knative/pkg/blob/0094d3a89242cddcd8532df4632fb619b2d7da09/kmeta/accessor.go#L47

Reply:
Done